filesmanager/filesmanager.py

```diff
@@ -570,16 +582,19 @@ def upload_file_to_directory(self, file, target_directory):
             from cms.djangoapps.contentstore.asset_storage_handler import \
                 update_course_run_asset  # pylint: disable=import-outside-toplevel
 
-        file_object = self.temporary_uploaded_files.pop(file.get("name"), None)
+        file_object = self.find_temporary_file(file.get("name"))
         file_path, name = file.get("path"), file.get("name")
         metadata = file.get("metadata", {})
 
         if file_object:
             file_path, name = self.generate_content_path(file_path, file_object.filename)
-            file_object.file._set_name(name) # pylint: disable=protected-access
+            file_object.file._set_name(self.generate_asset_name(file_path)) # pylint: disable=protected-access
```

Comment:
We need to update the name added in the `target_directory`

Reply:
No, that would cause long names to be displayed in the files manager xblock. It's better to keep a long internal name with the original name

Reply:
@BryanttV: what was the error this was causing?

Reply:
I'm not sure if the error you are getting is because of this but I'll describe it:

1. Upload a file in the Files Manager component in the Root, e.g. `Root/file.png`
2. Upload the same file from the course assets.

If we go back to the component the last file doesn't appear in the `Unpublished` folder.

Reply:
Can you test again? this behavior was not working correctly but now it's. Take a look: 

![image](https://github.com/eduNEXT/xblock-filesmanager/assets/33465240/e80a7411-d7b2-4f38-9ba5-2a4b8f99306b)